File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/PeerInfo.java
package com.shterneregen.securelan.common.model;

import java.util.Objects;

public record PeerInfo(
        String id,
        String username,
        String host,
        int port
) {
    public PeerInfo {
        Objects.requireNonNull(id, "id must not be null");
        Objects.requireNonNull(username, "username must not be null");
        Objects.requireNonNull(host, "host must not be null");
        if (port < 1 || port > 65535) {
            throw new IllegalArgumentException("port must be between 1 and 65535");
        }
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/PeerInfo.kt
package com.shterneregen.securelan.common.model

import java.util.Objects

@JvmRecord
data class PeerInfo(
    val id: String,
    val username: String,
    val host: String,
    val port: Int,
) {
    init {
        Objects.requireNonNull(id, "id must not be null")
        Objects.requireNonNull(username, "username must not be null")
        Objects.requireNonNull(host, "host must not be null")
        require(port in 1..65_535) { "port must be between 1 and 65535" }
    }
}
